flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBResourceContainer.java

```diff
@@ -207,4 +221,37 @@ public void close() throws Exception {
             sharedResources.close();
         }
     }
+
+    /**
+     * Overwrite configured {@link Filter} if enable partitioned filter. Partitioned filter only
+     * worked in full bloom filter, not blocked based.
+     */
+    private void overwriteFilterIfExist(BlockBasedTableConfig blockBasedTableConfig) {
```

Comment:
Suggest to directly overwrite the filter no matter what its original type is, since only full filters is a necessity to enable partitioned filter, and the `handlesToClose` collection will make sure the old filter will be closed w/o resource leak (only delayed).

What's more, it's hard to handle the case of failing to get filter from table config through reflection, we cannot let it through silently since the unknown filter type might be block based.

Reply:
I don't sure unused BloomFilter whether it consumes resources, so I close it manually. I'll revert this if it's ok to keep the useless filter.

Reply:
But still need use reflection verify is using filter, because can't access filter_ property directly.

Reply:
True, it's better not to set new filter if no filter exists at all. Then the suggested changes are:
1. Make `overwriteFilterIfExist` return a `boolean`
2. Let `getFilterFromBlockBasedTableConfig` throw out the reflection exception to distinguish it from the no filter existing case.
3. Make `overwriteFilterIfExist` return `true` if the filter is successfully overwritten or no filter exists, and return `false` if reflection exception thrown (we don't know whether block based filter is in use and have to disable partitioned index filter).

What do you think? @liuyufei9527 

```suggestion
    private boolean overwriteFilterIfExist(BlockBasedTableConfig blockBasedTableConfig) {
```